packages/ui/src/components/molecules/CouponInput.tsx

```diff
@@ -104,7 +114,7 @@ export const CouponInput: React.FC<CouponInputProps> = ({
           onClick={onApply}
           $isInvalid={isInvalid}
           // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-          disabled={disabled || isLoading}
+          disabled={disabled || isLoading || value.length !== 14}
```

Comment:
Take `14` as a prop and also prevent user from typing more characters than the limit